src/gobapi/dump/to_db.py

```diff
@@ -2,129 +2,254 @@
+        rename_table = _rename_table(self.schema, current_name=self.tmp_collection_name, new_name=self.collection_name)
+        result = self.engine.execute(rename_table)
+        result.close()
+
+    def _create_indexes(self, model):
+        """
+        Create default indexes for the given collection
+
+        :param model:
+        :return:
+        """
+        indexes = []
+        for field, spec in get_field_specifications(model).items():
+            if field == model['entity_id'] or re.compile(r"(^|.+_)ref$").match(field):
+                indexes.append(
+                    {'field': field})  # Plain entity id, full entity id (ref) or rel. foreign key (eg dst_ref)
+            elif "GOB.Geo" in spec['type']:
+                indexes.append({'field': field, 'method': "gist"})  # Spatial index
+            elif spec['type'] == "GOB.Reference" and "ref" in get_reference_fields(spec):
+                indexes.append({'field': f"{field}_ref"})  # Foreign key index
+
+        for index in indexes:
+            yield f"Create index on {index['field']}\n"
+            result = self.engine.execute(_create_index(self.schema, self.collection_name, **index))
+            result.close()
+
+    def _dump_entities_to_table(self, entities, model):
+        connection = self.engine.raw_connection()
+        stream = CSVStream(csv_entities(entities, model), STREAM_PER)
+
+        with connection.cursor() as cursor:
+            yield "Export data"
+            commit = COMMIT_PER
+            while stream.has_items():
+                stream.reset_count()
+                cursor.copy_expert(
+                    sql=f"COPY {self.schema}.{self.tmp_collection_name} FROM STDIN DELIMITER ';' CSV HEADER;",
+                    file=stream,
+                    size=BUFFER_PER
+                )
+
+                if stream.total_count >= commit:
+                    connection.commit()
+                    commit += COMMIT_PER
+
+                    yield f"\n{self.collection_name}: {stream.total_count:,}"
+                else:
+                    # Let client know we're still working.
+                    yield "."
+
+        yield f"\nExported {stream.total_count} rows\n"
+        connection.commit()
+
+    def _filter_last_events_lambda(self, max_eventid):
+        return lambda table: getattr(table, FIELD.LAST_EVENT) > max_eventid
+
+    def dump_to_db(self, full_dump=False):
+        """Runs dump for this instance. Tries to synchronise based on last event by default. Set full_dump=True to
+        replace all existing data and ignore synchronisation based on events.
+        """
+        yield from self._prepare_destination()
+
+        max_eventid = None if full_dump else self._max_eventid_dst()
+
+        if max_eventid is not None and self._table_columns_equal(self.collection_name, self.tmp_collection_name):
+            yield "Have earlier dump and columns have not changed. Do sync dump\n"
```

Comment:
Can we add a test if max_eventid <= "current max eventid" or do you like to postpone this to a new PR?
In case that max_eventid > current than we can issue an error, truncate the dst table and perform a full dump

Reply:
Or raise an exception because it should (at least for now) never be possible

Reply:
Forcing a full dump now when this happens.